File: backend/app/schemas/resume.py
"""Resume Pydantic schemas."""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ResumeResponse(BaseModel):
    """Resume response schema."""

    id: uuid.UUID
    filename: str
    content_type: str
    file_size: int
    is_primary: bool
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ResumeListResponse(BaseModel):
    """List of resumes."""

    resumes: list[ResumeResponse]
    total: int

File: frontend/types/resume.ts
/**
 * TypeScript types for resumes.
 */

export interface ResumeResponse {
  id: string;
  filename: string;
  content_type: string;
  file_size: number;
  is_primary: boolean;
  created_at: string;
}

export interface ResumeListResponse {
  resumes: ResumeResponse[];
  total: number;
}
